File: Study Plans/Binary Search I/Day 1/374. Guess Number Higher or Lower.js
/** 
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */

/**
 * @param {number} n
 * @return {number}
 */
var guessNumber = function(n) {
    let l = 1, h = n;
    while(l <= h) {
        let mid = Math.floor((h+l)/2);
        if(guess(mid) == 0) return mid;
        else if(guess(mid) == -1) h = mid-1;
        else if(guess(mid) == 1) l = mid+1;
    }
    return n;
};

File: LeetCode/CPP/Easy/374. Guess Number Higher or Lower.cpp
// Problem Link:
// https://leetcode.com/problems/guess-number-higher-or-lower/

// Solution:
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        int m = n/2, pick;
        while(true) {
            if(guess(m) == 0) {
                pick = m;
                break;
            }
            else if(guess(m) == 1) {
                m++;
            }
            else {
                m--;
            }
        }
        return pick;
    }
};